app/actions/async.js

```diff
@@ -671,16 +665,26 @@ export function checkUploadTargetUserAndMaybeRedirect() {
       return _.includes(supportedDeviceKeys, key);
     });
     if (_.isEmpty(targetedDevices) || !atLeastOneDeviceSupportedOnSystem) {
-      return dispatch(syncActions.setPage(pages.SETTINGS, undefined, {metric: {eventName: metrics.CLINIC_NEXT}}));
+      return dispatch(setPage(pages.SETTINGS, undefined, {metric: {eventName: metrics.CLINIC_NEXT}}));
     } else {
-      return dispatch(syncActions.setPage(pages.MAIN, undefined, {metric: {eventName: metrics.CLINIC_NEXT}}));
+      return dispatch(setPage(pages.MAIN, undefined, {metric: {eventName: metrics.CLINIC_NEXT}}));
     }
   };
 }
 
 export function clickAddNewUser(){
   return (dispatch, getState) =>{
     dispatch(syncActions.setUploadTargetUser(null));
-    dispatch(syncActions.setPage(pages.CLINIC_USER_EDIT, undefined, {metric: {eventName: metrics.CLINIC_ADD}}));
+    dispatch(setPage(pages.CLINIC_USER_EDIT, undefined, {metric: {eventName: metrics.CLINIC_ADD}}));
+  };
+}
+
+export function setPage(page, actionSource = actionSources[actionTypes.SET_PAGE], metric) {
```

Comment:
The new setPage function. This is a thin wrapper around the `react-redux-router` push function which creates an action to navigate to a new page (pushes the path to the browser history object) which also tacks on our meta/metric information into the location change `state` property, from which it is retrieved in middleware for tracking.